XMonad/Layout/BinarySpacePartition.hs

```diff
@@ -108,37 +109,48 @@ import Data.Ratio ((%))
 -- > , ((myModMask .|. shiftMask, xK_a),     sendMessage Equalize)
 --
 
--- |Message for rotating the binary tree around the parent node of the window to the left or right
+-- | Message for rotating the binary tree around the parent node of the window to the left or right
 data TreeRotate = RotateL | RotateR deriving Typeable
 instance Message TreeRotate
 
--- |Message to balance the tree in some way (Balance retiles the windows, Equalize changes ratios)
+-- | Message to balance the tree in some way (Balance retiles the windows, Equalize changes ratios)
 data TreeBalance = Balance | Equalize deriving Typeable
 instance Message TreeBalance
 
--- |Message for resizing one of the cells in the BSP
-data ResizeDirectional = ExpandTowards Direction2D | ShrinkFrom Direction2D | MoveSplit Direction2D deriving Typeable
+-- | Message for resizing one of the cells in the BSP
+data ResizeDirectional =
+        ExpandTowardsBy Direction2D Rational
+      | ShrinkFromBy Direction2D Rational
+      | MoveSplitBy Direction2D Rational deriving Typeable
 instance Message ResizeDirectional
 
--- |Message for rotating a split (horizontal/vertical) in the BSP
+-- | Patterns to replicate the old messages
```

Comment:
This comment is just for `ExpandTowards`. Please add a comment to each pattern, and say what it's equivalent to (e.g. `ExpandTowards L` is equivalent to `ExpandTowardsBy L 0.05`